### PR TITLE
try using GITHUB_TOKEN in CI and obtaining fewer 429s etc?

### PR DESCRIPTION
**Pull request by mlinksva**

### CODE REVIEW

**Comment by nschonni** [comment]
Probably need to do this in the `scripts` too

**Comment by mlinksva** [comment]
Yeah, will add elsewhere if/when this works.

File: Rakefile [hunk 1]
```diff
@@ -4,9 +4,14 @@ require 'html-proofer'
...
-                              hydra: { max_concurrency: 10 },
-                              url_ignore: [%r{https://developer.github.com}, %r{https://docs.github.com}, %r{https://help.github.com}],
-                              http_status_ignore: [429]).run
+  proofer = HTMLProofer.check_directory(
+    './_site/',
+    check_html: true,
+    hydra: { max_concurrency: 10 },
+    url_ignore: [%r{https://developer.github.com}, %r{https://docs.github.com}, %r{https://help.github.com}],
+    http_status_ignore: [429])
+  proofer.before_request do |request|
```
Comment (nschonni) [inline Rakefile]: might want to check if the value is set, so the local script works if they haven't set this up. Maybe the empty header isn't an issue though

File: Rakefile [hunk 2]
```diff
@@ -4,9 +4,14 @@ require 'html-proofer'
...
+    request.options[:headers]['Authorization'] = "Bearer #{GITHUB_TOKEN}" if request.base_url == "https://github.com"
```
Comment (nschonni) [inline Rakefile]: I'm bad at ruby, but does it need to be something like `ENV["GITHUB_TOKEN"]`

Comment (mlinksva) [reply Rakefile]: Me too, just noticed due to failing CI :)

**Comment by mlinksva** [comment]
Mysteries:
* obviously not avoiding 429
* ~~htmlproofer config should ingore them~~ (updated htmlproofer config for new version)
* ~~Not sure why 2 workflows aren't running~~ (they only run if changes in `_data`, by design!)

**Comment by nschonni** [comment]
Other approach might be to try getting it working on one of the other scripts with the native Typeous config, then use that for HTMLProofer https://github.com/gjtorikian/html-proofer/wiki/How-to-configure-Typhoeus

**Comment by mlinksva** [comment]
Not clear to me passing token is working, but other fixes here useful. Tests pass for 1st time in awhile.

**Pull request merged**